prov/tcp/src/tcpx_init.c

```diff
@@ -122,12 +122,22 @@ struct tcpx_port_range port_range = {
 	.high = 0,
 };
 
-static void tcpx_init_env(void)
+static int tcpx_init_env(void)
 {
 	srand(getpid());
 
 	fi_param_get_int(&tcpx_prov, "port_high_range", &port_range.high);
 	fi_param_get_int(&tcpx_prov, "port_low_range", &port_range.low);
+
+	if (port_range.high > TCPX_PORT_MAX_RANGE) {
+		port_range.high = TCPX_PORT_MAX_RANGE;
+	}
+	if (port_range.low < 0 || port_range.high < 0
+			       || port_range.low > port_range.high) {
+		FI_WARN(&tcpx_prov, FI_LOG_EP_CTRL,"Invalid info\n");
```

Comment:
More descriptive warning would be helpful here. Something like "User provided port range invalid"

Reply:
@vkrishna Done.